force-app/main/default/classes/EncryptionRecipes.cls

```diff
@@ -0,0 +1,414 @@
+        String blobsAsHex = EncodingUtil.convertToHex(dataToDecrypt);
+        // Substract first two bytes (initialization vector)
+        String initializationVectorString = blobsAsHex.substring(0, 32);
+        // Substract the rest (encrypted data)
+        String encryptedDataString = blobsAsHex.substring(32);
+        // Convert both hex values to Blob
+        Blob initializationVector = EncodingUtil.convertFromHex(
+            initializationVectorString
+        );
+        Blob encryptedData = EncodingUtil.convertFromHex(encryptedDataString);
+        // Call Crypto.decryptWithManagedIV specifying the selected AES Algorithm
+        return Crypto.decrypt(
+            EncryptionRecipes.AESAlgorithm.AES256.name(),
+            AES_KEY,
+            initializationVector,
+            encryptedData
+        );
+    }
+
+    /**
+     * @description Aux method to generate a random initialization vector.
+     * @return Blob
+     **/
+    public static Blob generateInitializationVector() {
+        String characters = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789abcdefghijklmnopqrstuvwxyz';
+        String randomString = '';
+        while (randomString.length() < 16) {
+            Integer index = Math.mod(
+                Math.abs(Crypto.getRandomInteger()),
+                characters.length()
+            );
+            randomString += characters.substring(index, index + 1);
+        }
+        return Blob.valueOf(randomString);
+    }
+
+    /**
+     * @description Generates one-way hash digest that can be checked in destination to ensure integrity.
+     * @param dataToHmac Blob that contains some data for which to generate a hash
+     * @return Blob
+     * @example
+     * Blob dataToHash = Blob.valueOf('Test data');
+     * Blob hash = EncryptionRecipes.generateHashRecipe();
+     * System.debug(EncodingUtil.base64Encode(hash));
+     **/
+    @AuraEnabled
+    public static Blob generateHashRecipe(Blob dataToHash) {
+        // Call Crypto.generateDigest specifying the selected algorithm
+        return Crypto.generateDigest(HashAlgorithm.SHA512.name(), dataToHash);
+    }
+
+    /**
+     * @description Recomputes hash digest for and compares it with the received one, throwing an exception if they're not equal.
+     * @param hash Blob that contains the received hash
+     * @param dataToCheck Blob that contains the data to check the hash for
+     * @return void
+     * @example
+     * try {
+     *  EncryptionRecipes.checkHashRecipe(hash, corruptedData);
+     * } catch(Exception e) {
+     *  // Should log exception
+     *  System.debug(e.getMessage());
+     * }
+     **/
+    @AuraEnabled
+    public static void checkHashRecipe(Blob hash, Blob dataToCheck) {
+        Blob recomputedHash = Crypto.generateDigest(
+            HashAlgorithm.SHA512.name(),
+            dataToCheck
+        );
+
+        // recomputedHash and hash should  be identical!
+        if (hash != recomputedHash) {
```

Comment:
Comparisons which involve cryptography are performed in constant time using specialized functions.[1][2]. The real world implications will depend on usage [3].

[1] - https://en.wikipedia.org/wiki/Timing_attack
[2] - https://github.com/bcgit/bc-java/blob/bc3b92f1f0e78b82e2584c5fb4b226a13e7f8b3b/tls/src/main/java/org/bouncycastle/tls/TlsUtils.java#L1454
[3] - https://www.blackhat.com/docs/us-14/materials/us-14-Mayer-Time-Trial-Racing-Towards-Practical-Timing-Attacks.pdf (slide 47 and on)